libs/community/langchain_community/chat_models/naver.py

```diff
@@ -0,0 +1,565 @@
+
+    ncp_apigw_api_key: Optional[SecretStr] = Field(default=None, alias="apigw_api_key")
+    """Automatically inferred from env are `NCP_APIGW_API_KEY` if not provided."""
+
+    base_url: Optional[str] = Field(default=None, alias="base_url")
+    """
+    Automatically inferred from env are `NCP_CLOVASTUDIO_API_BASE_URL` if not provided.
+    """
+
+    temperature: Optional[float] = None
+    top_k: Optional[int] = None
+    top_p: Optional[float] = None
+    repeat_penalty: Optional[float] = None
+    max_tokens: Optional[int] = None
+    stop_before: Optional[list[str]] = Field(default=None, alias="stop")
+    include_ai_filters: Optional[bool] = None
+    seed: Optional[int] = None
+
+    timeout: Optional[int] = 90
+    max_retries: Optional[int] = 2
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        allow_population_by_field_name = True
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        """Get the default parameters for calling the API."""
+        defaults = {
+            "temperature": self.temperature,
+            "topK": self.top_k,
+            "topP": self.top_p,
+            "repeatPenalty": self.repeat_penalty,
+            "maxTokens": self.max_tokens,
+            "stopBefore": self.stop_before,
+            "includeAiFilters": self.include_ai_filters,
+            "seed": self.seed,
+        }
+        filtered = {k: v for k, v in defaults.items() if v is not None}
+        return filtered
+
+    @property
+    def _identifying_params(self) -> Dict[str, Any]:
+        """Get the identifying parameters."""
+        self._default_params["model_name"] = self.model_name
+        return self._default_params
+
+    @property
+    def lc_secrets(self) -> Dict[str, str]:
+        return {
+            "ncp_clovastudio_api_key": "NCP_CLOVASTUDIO_API_KEY",
+            "ncp_apigw_api_key": "NCP_APIGW_API_KEY",
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of chat model."""
+        return "chat-naver"
+
+    def _get_ls_params(
+        self, stop: Optional[List[str]] = None, **kwargs: Any
+    ) -> LangSmithParams:
+        """Get the parameters used to invoke the model."""
+        params = super()._get_ls_params(stop=stop, **kwargs)
+        params["ls_provider"] = "naver"
+        return params
+
+    @property
+    def _client_params(self) -> Dict[str, Any]:
+        """Get the parameters used for the client."""
+        return self._default_params
+
+    @property
+    def _api_url(self) -> str:
+        """GET chat completion api url"""
+        app_type = "serviceapp" if self.service_app else "testapp"
+
+        if self.task_id:
+            return (
+                f"{self.base_url}/{app_type}/v1/tasks/{self.task_id}/chat-completions"
+            )
+        else:
+            return f"{self.base_url}/{app_type}/v1/chat-completions/{self.model_name}"
+
+    def __init__(self, **kwargs: Any) -> None:
```

Comment:
let's make these pydantic validated fields instead - much cleaner code! e.g. for numeric fields: https://docs.pydantic.dev/latest/concepts/fields/#numeric-constraints

Reply:
ok I'll change to pydantic validated fields on tomorrow(9/24).

Reply:
Applied.